custom/icds_reports/views.py

```diff
@@ -137,3 +154,338 @@ def get_tableau_access_token(tableau_user, client_ip):
+
+
+@location_safe
+@method_decorator([login_and_domain_required], name='dispatch')
+class AwcReportsView(View):
+    def get(self, request, *args, **kwargs):
+        step = kwargs.get('step')
+
+        now = datetime.utcnow()
+        month_param = int(request.GET.get('month', now.month))
+        year_param = int(request.GET.get('year', now.year))
+        month = datetime(year_param, month_param, 1)
+        prev_month = month - relativedelta(months=1)
+        three_month = month - relativedelta(months=2)
+        location = request.GET.get('location', None)
+        aggregation_level = 5
+
+        config = {
+            'aggregation_level': aggregation_level,
+            'awc_site_code': "awc_214"
+        }
+        if location:
+            try:
+                sql_location = SQLLocation.objects.get(location_id=location, domain=self.kwargs['domain'])
+                location_key = '%s_site_code' % sql_location.location_type.code
+                config.update({
+                    location_key: sql_location.site_code,
+                })
+            except SQLLocation.DoesNotExist:
+                pass
+
+        data = []
+        if step == 'system_usage':
+            data = get_awc_reports_system_usage(
+                config,
+                tuple(month.timetuple())[:3],
+                tuple(prev_month.timetuple())[:3],
+                tuple(three_month.timetuple())[:3],
+                'aggregation_level'
+            )
+        elif step == 'pse':
+            data = get_awc_reports_pse(
+                config,
+                tuple(month.timetuple())[:3],
+                tuple(three_month.timetuple())[:3]
+            )
+        elif step == 'maternal_child':
+            data = get_awc_reports_maternal_child(
+                config,
+                tuple(month.timetuple())[:3],
+                tuple(three_month.timetuple())[:3]
+            )
+        elif step == 'demographics':
+            data = get_awc_report_demographics(
+                config,
+                tuple(month.timetuple())[:3]
+            )
+        elif step == 'beneficiary':
+            data = get_awc_report_beneficiary(
+                config['awc_site_code'],
+                tuple(month.timetuple())[:3],
+                tuple(three_month.timetuple())[:3],
+            )
+        return JsonResponse(data=data)
+
+
+@method_decorator([login_and_domain_required], name='dispatch')
+class ExportIndicatorView(View):
+    def post(self, request, *args, **kwargs):
+
+        export_format = request.POST.get('format')
+        month = int(request.POST.get('month'))
+        year = int(request.POST.get('year'))
+        aggregation_level = int(request.POST.get('aggregation_level'))
+        indicator = int(request.POST.get('indicator'))
+
+        config = {
+            'aggregation_level': aggregation_level
+        }
+
+        if month and year:
+            config.update({
+                'month': datetime(year, month, 1).date(),
```

Comment:
nit: should construct with a `date(year, month,day)` instead of `datetime(year, month, day).date()`